app/javascript/js/controllers/tabs_controller.js

```diff
@@ -70,4 +79,141 @@ export default class extends Controller {
   hideAllTabs() {
     this.tabPanelTargets.map((element) => element.classList.add('hidden'))
   }
+
+  updateActiveTabButtons(tabName) {
+    if (!this.hasTabButtonTarget) return
+
+    this.tabButtonTargets.forEach((button) => {
+      const isActive = button.dataset.tabId === tabName
+      const isFocused = document.activeElement === button
+
+      button.setAttribute('aria-selected', isActive)
+
+      if (isActive || isFocused) {
+        button.setAttribute('tabindex', '0')
+      } else {
+        button.setAttribute('tabindex', '-1')
+      }
+
+      this.toggleTabClasses(button, isActive)
+
+      const wrapper = button.closest('.tabs__item-wrapper--scope')
+      if (wrapper) {
+        this.toggleTabClasses(wrapper, isActive, button)
+      }
+    })
+  }
+
+  toggleTabClasses(element, isActive, sourceElement = element) {
+    const activeClass = sourceElement.dataset.tabActiveClass
+    const inactiveClass = sourceElement.dataset.tabInactiveClass
+
+    if (activeClass) element.classList.toggle(activeClass, isActive)
+    if (inactiveClass) element.classList.toggle(inactiveClass, !isActive)
+  }
+
+  handleKeyDown(e) {
+    const currentTab = e.currentTarget
+    const tabs = this.getEnabledTabs()
+    const currentIndex = tabs.indexOf(currentTab)
+
+    if (currentIndex === -1) return
+
+    let targetTab = null
+    let shouldPreventDefault = false
+
+    switch (e.key) {
+      case 'ArrowRight':
+      case 'ArrowDown':
+        targetTab = tabs[(currentIndex + 1) % tabs.length]
+        shouldPreventDefault = true
+        break
+
+      case 'ArrowLeft':
+      case 'ArrowUp':
+        targetTab = tabs[(currentIndex - 1 + tabs.length) % tabs.length]
+        shouldPreventDefault = true
+        break
+
+      case 'Home':
+        targetTab = tabs[0]
+        shouldPreventDefault = true
+        break
+
+      case 'End':
+        targetTab = tabs[tabs.length - 1]
+        shouldPreventDefault = true
+        break
+
+      case 'Enter':
+      case ' ':
+        if (currentTab.dataset.tabId) {
+          this.activateTab(currentTab)
+          shouldPreventDefault = true
+        }
+        break
+    }
+
+    if (shouldPreventDefault) {
+      e.preventDefault()
+    }
+
+    if (targetTab) {
+      this.focusTab(targetTab)
+    }
+  }
+
+  getEnabledTabs() {
+    return this.tabButtonTargets.filter((button) => {
+      const disabled =        button.getAttribute('aria-disabled') === 'true'
+        || button.hasAttribute('disabled')
+
+      return !disabled
+    })
+  }
+
+  focusTab(tabButton) {
+    // Remove focus class from all tabs first
+    this.removeKeyboardFocusClass()
+
+    // Add keyboard focus class and focus the tab
+    tabButton.classList.add('tabs__item--focused')
+    tabButton.setAttribute('tabindex', '0')
+    tabButton.focus()
+  }
```

Comment:
I think this method does too much.
I mean, when you hit tab it already cycles through  those tabs automatically focusing the right one.
I feel it tries to reinvent what the browser already does, right?